src/ast/nodes/VariableDeclaration.ts

```diff
@@ -175,13 +199,28 @@ export default class VariableDeclaration extends NodeBase {
 			} else {
 				if (options.format === 'system' && node.init !== null) {
 					if (node.id.type !== NodeType.Identifier) {
-						node.id.addExportedVariables(systemPatternExports);
-					} else if (node.id.variable!.exportName) {
-						code.prependLeft(
-							code.original.indexOf('=', node.id.end) + 1,
-							` exports('${node.id.variable!.safeExportName || node.id.variable!.exportName}',`
-						);
-						nextSeparatorString += ')';
+						node.id.addExportedVariables(systemPatternExports, options.exportNamesByVariable);
+					} else {
+						const exportNames = options.exportNamesByVariable.get(node.id.variable!);
+						if (exportNames) {
+							const _ = options.compact ? '' : ' ';
+							const operatorPos = findFirstOccurrenceOutsideComment(
+								code.original,
+								'=',
+								node.id.end
+							);
+							const nextIsSpace = code.original[operatorPos + 1] === ' ';
```

Comment:
Similar, I think this is adding a lot of complexity to a really complicated function and would hope we can get rid of this, even if the output is not as nice. Also what about more spaces? Tabs? etc.